process/accountProcessor.go

```diff
@@ -162,6 +162,36 @@ func (ap *AccountProcessor) GetAllESDTTokens(address string) (*data.GenericAPIRe
 	return nil, ErrSendingRequest
 }
 
+// GetKeyValuePairs returns all the key-value pairs for a given address
+func (ap *AccountProcessor) GetKeyValuePairs(address string) (*data.GenericAPIResponse, error) {
+	observers, err := ap.getObserversForAddress(address)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, observer := range observers {
+		apiResponse := data.GenericAPIResponse{}
+		apiPath := AddressPath + address + "/keys"
+		respCode, err := ap.proc.CallGetRestEndPoint(observer.Address, apiPath, &apiResponse)
+		if err == nil || respCode == http.StatusBadRequest || respCode == http.StatusInternalServerError {
```

Comment:
Is this condition ok? Maybe you can extract it in one explanatory variable.

Reply:
Yes. It is correct. If you look inside `CallGetRestEndPoint` function it can return values on both return params (the response code and the error) so this check must be made in order to return if a request isn't 200 OK.

I tried to extract a variable, but I would get an IDE's linter issue that tells me the `err` can be nil when logging it. I would leave it as it is

Reply:
ok